packages/deno/src/integrations/denoserver.ts

```diff
@@ -0,0 +1,136 @@
+import {
+  SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN,
+  SEMANTIC_ATTRIBUTE_SENTRY_SOURCE,
+  captureException,
+  continueTrace,
+  defineIntegration,
+  setHttpStatus,
+  startSpan,
+  withIsolationScope,
+} from '@sentry/core';
+import type { IntegrationFn, SpanAttributes } from '@sentry/types';
+import { getSanitizedUrlString, parseUrl } from '@sentry/utils';
+
+type RawHandler = (request: Request, info: Deno.ServeHandlerInfo) => Response | Promise<Response>;
+
+const INTEGRATION_NAME = 'DenoServer';
+
+const _denoServerIntegration = (() => {
+  return {
+    name: INTEGRATION_NAME,
+    setupOnce() {
+      instrumentDenoServe();
+    },
+  };
+}) satisfies IntegrationFn;
+
+/**
+ * Instruments `Deno.serve` to automatically create transactions and capture errors.
+ *
+ * Enabled by default in the Deno SDK.
+ *
+ * ```js
+ * Sentry.init({
+ *   integrations: [
+ *     Sentry.denoServerIntegration(),
+ *   ],
+ * })
+ * ```
+ */
+export const denoServerIntegration = defineIntegration(_denoServerIntegration);
+
+/**
+ * Instruments Deno.serve by patching it's options.
+ */
+export function instrumentDenoServe(): void {
+  Deno.serve = new Proxy(Deno.serve, {
+    apply(serveTarget, serveThisArg, serveArgs: unknown[]) {
+      const [arg1, arg2] = serveArgs;
+
+      if (typeof arg1 === 'function') {
+        serveArgs[0] = instrumentDenoServeOptions(arg1 as RawHandler);
+      } else if (typeof arg2 === 'function') {
+        serveArgs[1] = instrumentDenoServeOptions(arg2 as RawHandler);
+      } else if (arg1 && typeof arg1 === 'object' && 'handler' in arg1 && typeof arg1.handler === 'function') {
+        arg1.handler = instrumentDenoServeOptions(arg1.handler as RawHandler);
+      }
+
+      return serveTarget.apply(serveThisArg, serveArgs as Parameters<typeof Deno.serve>);
+    },
+  });
+}
+
+/**
+ * Instruments Deno.serve handler to automatically create spans and capture errors.
+ */
+function instrumentDenoServeOptions(handler: RawHandler): RawHandler {
+  return new Proxy(handler, {
+    apply(handlerTarget, handlerThisArg, handlerArgs: Parameters<RawHandler>) {
+      return withIsolationScope(isolationScope => {
+        isolationScope.clear();
```

Comment:
also this line should be removed when https://github.com/getsentry/sentry-javascript/issues/12450#issuecomment-2161141286 is resolved